### DISCUSSION TITLE
where is the logic process the open workscpace ?

### DISCUSSION THREAD

**Discussion by icoco** [eclipse-theia/theia#13817] [Q&A]
eg, while theia browser start, navigate to a folder : http://127.0.0.1:3000/#/abc/xyz

 no express route handle the path '/#' request,  I am wondering which module process the online request .

again, if have plan provide standard restful API to open the workspace , eg :
```
 http://127.0.0.1:3000/api/workspace/{directory path}  (open workspace)

 http://127.0.0.1:3000/api/workspace/{directory path}?file=xxx (open workspace and open special file )
```
I guess that current version has such ability but seem no document guideline  yet

**Comment by msujew** [comment]
Hey @icoco,

the [`WorkspaceService`](https://github.com/eclipse-theia/theia/blob/master/packages/workspace/src/browser/workspace-service.ts) in the browser part of the `@theia/workspace` package is responsible for handling which workspace to open.
